### PR TITLE
docs: Add how-to guide to create Intel Notebook 

### LINKED ISSUE #149: Write `dss` doc on how to spin up Notebook from Intel with IPEX or ITEX
### Why it needs to get done

Write a documentation describing that the user can simply deploy intel notebooks with following commands (this is only possible when the device plugin is enabled otherwise the notebooks will be deployed but resources will not be available).

```
dss create my-itex-notebook --image=intel/intel-extension-for-tensorflow:2.15.0-xpu-idp-jupyter
dss create my-ipex-notebook --image=intel/intel-extension-for-pytorch:2.1.20-xpu-idp-jupyte
```

### What needs to get done

- Write a documentation about how to dpeloy Intel images on DSS.

### When is the task considered done

- Documentation is written and tested.

### PR DESCRIPTION
**Pull request by NohaIhab**
closes #149 

Note to reviewer: I'm referencing the guide from PR #160 that is not merged yet so the documentation build is expected to fail

### CODE REVIEW

**Comment by afgambin** [comment]
lgtm @NohaIhab. Let's wait for merging before the other PR. One comment, can we add any confirmation/verification the notebook has been properly created? as we do for the NVIDIA case

**Comment by NohaIhab** [comment]
Hi @afgambin, I dropped the Intel team a question to ask if such command exists with which we can verify the same way we do with NVIDIA. In case it is not possible, we can instead link the calculation example Michal is working on in https://github.com/canonical/data-science-stack/issues/150 telling users to try running it.

**Comment by afgambin** [comment]
@NohaIhab the PR from Intel GPU has been merged. We can link it now from here.

**Comment by afgambin** [comment]
It was actually better to keep the previous way of linking using `:ref:` @NohaIhab.

File: docs/how-to/jupyter-notebook.rst [hunk 1]
```diff
@@ -83,6 +83,52 @@ Confirm the GPU is detected and usable by running:
 
    tf.config.list_physical_devices('GPU')
 
+Create an Intel GPU-enabled notebook
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can create an Intel GPU-enabled Jupyter Notebook with IPEX or ITEX.
```
Comment (afgambin) [inline docs/how-to/jupyter-notebook.rst:89]: Can we add a link under "IPEX" and "ITEX" for referencing the user to further info?

File: docs/how-to/jupyter-notebook.rst [hunk 2]
```diff
@@ -83,6 +83,52 @@ Confirm the GPU is detected and usable by running:
...
+
+.. note::
+
+   To launch an Intel GPU-enabled notebook, you must first :doc:`Enable Intel GPUs <../how-to/enable-gpus/enable-intel-gpu>`.
```
Comment (afgambin) [inline docs/how-to/jupyter-notebook.rst:93]: Let's use `:ref:` + the tag here for the internal linking. I committed this change but probably you didn't refresh your local PR with my change and got overwritten.

File: docs/how-to/jupyter-notebook.rst [hunk 3]
```diff
@@ -83,6 +83,52 @@ Confirm the GPU is detected and usable by running:
...
+        - intel-pytorch = intel/intel-extension-for-pytorch:2.1.20-xpu-idp-jupyter
+        - intel-tensorflow = intel/intel-extension-for-tensorflow:2.15.0-xpu-idp-jupyter
+
+Select one of them and create a notebook as follows:
+
+.. code-block:: bash
+
+   dss create my-itex-notebook --image=intel-tensorflow
+
+Confirm the devices are detected and usable by running:
```
Comment (afgambin) [inline docs/how-to/jupyter-notebook.rst:114]:
```suggestion
Confirm the GPU is detected and usable by running:
```
Let's focus on the GPU, not devices in general

File: docs/how-to/jupyter-notebook.rst [hunk 4]
```diff
@@ -83,6 +83,52 @@ Confirm the GPU is detected and usable by running:
...
+
+.. code-block:: python
+
+   import tensorflow as tf
+
+   tf.config.experimental.list_physical_devices()
+
+Example of the expected output for a host system containing an Intel CPU and a single Intel GPU:
```
Comment (afgambin) [inline docs/how-to/jupyter-notebook.rst:122]:
```suggestion
For example, you should expect an output like the following for a host system containing an Intel CPU and a single Intel GPU:
```

File: docs/how-to/jupyter-notebook.rst [hunk 5]
```diff
@@ -83,6 +83,52 @@ Confirm the GPU is detected and usable by running:
...
+
+.. code-block:: python
+
+    [PhysicalDevice(name='/physical_device:CPU:0', device_type='CPU'), PhysicalDevice(name='/physical_device:XPU:0', device_type='XPU')]
+
+.. note::
+
+    Intel calls GPU+CPU mode XPU
```
Comment (afgambin) [inline docs/how-to/jupyter-notebook.rst:130]:
```suggestion
    Note that Intel denotes XPU the combination of an Intel CPU with GPU.
```

**Comment by afgambin** [comment]
Did a pass @NohaIhab, left you some suggestions.

File: docs/how-to/jupyter-notebook.rst (hunk 4 above)
Comment (NohaIhab) [reply docs/how-to/jupyter-notebook.rst]: done

File: docs/how-to/jupyter-notebook.rst (hunk 3 above)
Comment (NohaIhab) [reply docs/how-to/jupyter-notebook.rst]: done

File: docs/how-to/jupyter-notebook.rst (hunk 1 above)
Comment (NohaIhab) [reply docs/how-to/jupyter-notebook.rst:89]: posted the question on the MM channel, not sure which links we can use here

File: docs/how-to/jupyter-notebook.rst (hunk 2 above)
Comment (NohaIhab) [reply docs/how-to/jupyter-notebook.rst]: done

File: docs/how-to/jupyter-notebook.rst (hunk 5 above)
Comment (NohaIhab) [reply docs/how-to/jupyter-notebook.rst]: done

**Review by afgambin — APPROVED** [review]
lgtm @NohaIhab, good job!

**Pull request merged**